src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java

```diff
@@ -666,4 +676,85 @@ public static boolean supportJDBC43(Connection con) throws SQLException {
     public static String escapeSingleQuotes(String name) {
         return name.replace("'", "''");
     }
-}
+    
+    /**
+     * Returns if connected to SQL Azure
+     * 
+     * @param con
+     *        connection to server
+     * @return boolean
+     * @throws SQLException
+     */
+    public static boolean isSqlAzure(Connection con) throws SQLException {
+        if (_determinedSqlAzureOrSqlServer) {
+            return _isSqlAzure;
+        }
+
+        try {
+            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+        } catch (ClassNotFoundException e) {
+            throw new SQLException(e.toString());
+        }
+
+        boolean ownsCon = false;
+
+        if (null == con) {
+            con = DriverManager.getConnection(AbstractTest.getConnectionString());
```

Comment:
Why can we not own the connection in the method itself and close it in the end? I don't understand the point of `ownCon` property in this class.. even it is implemented for other reasons, doesn't have to be necessarily used in new methods that we add here.

Reply:
This is faster as we don't always have to create a new connection if the method that's calling isSqlAzure already has access to a connection. However, since this is a static method that can be called anywhere, there's places where the caller method doesn't have access to their own connection, in which case this method creates/closes a new connection. It's mainly here for better performance.